### ISSUE TITLE
[BUG] bloxstrap gets stuck on applying Roblox modifications

### ISSUE THREAD

**Issue by gunnadademon93** [bloxstraplabs/bloxstrap#3492]
### Acknowledgement of preliminary instructions

- [X] I have read the preliminary instructions, and I am certain that my problem has not already been addressed.
- [X] I have thoroughly looked through the available Wiki articles and could not find a solution to my problem.
- [X] I am using the latest version of Bloxstrap.
- [ ] I did not answer truthfully to all the above checkboxes.

### Bloxstrap Version

v2.8.0

### What problem did you encounter?

It won’t open my bloxstrap

### Bloxstrap Log

```text
2024-10-27T00:53:53Z [App::OnStartup] Starting Bloxstrap v2.8.0
2024-10-27T00:53:53Z [App::OnStartup] Compiled Wednesday, 16 October 2024 at 5:44:37 AM from commit 42ffec26cf2e080fe22210305a08fcb286aeb577 (tag/v2.8.0)
2024-10-27T00:53:53Z [App::OnStartup] Loaded from %UserProfile%\Downloads\Bloxstrap-v2.8.0.exe
2024-10-27T00:53:53Z [App::OnStartup] Temp path is %UserProfile%\AppData\Local\Temp\Bloxstrap
2024-10-27T00:53:53Z [App::OnStartup] WindowsStartMenu path is %UserProfile%\AppData\Roaming\Microsoft\Windows\Start Menu\Programs
2024-10-27T00:53:53Z [Logger::Initialize] Initializing at %UserProfile%\AppData\Local\Bloxstrap\Logs\Bloxstrap_20241027T005353Z.log
2024-10-27T00:53:53Z [Logger::Initialize] Finished initializing!
2024-10-27T00:53:53Z [JsonManager<Settings>::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\Settings.json...
2024-10-27T00:53:53Z [JsonManager<Settings>::Load] Loaded successfully!
2024-10-27T00:53:53Z [JsonManager<State>::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\State.json...
2024-10-27T00:53:53Z [JsonManager<State>::Load] Loaded successfully!
2024-10-27T00:53:53Z [FastFlagManager::Load] Loading from %UserProfile%\AppData\Local\Bloxstrap\Modifications\ClientSettings\ClientAppSettings.json...
2024-10-27T00:53:53Z [FastFlagManager::Load] Loaded successfully!
2024-10-27T00:53:53Z [FastFlagManager::SetValue] Setting of 'FLogNetwork' to '7' is pending
2024-10-27T00:53:56Z [LaunchHandler::LaunchRoblox] Initializing bootstrapper
2024-10-27T00:53:56Z [LaunchHandler::LaunchRoblox] Initializing bootstrapper dialog
2024-10-27T00:53:56Z [Bootstrapper::Run] Running bootstrapper
2024-10-27T00:53:56Z [Bootstrapper::SetStatus] Connecting to Roblox...
2024-10-27T00:53:57Z [Deployment::TestConnection<https://setup.rbxcdn.com>] Connecting...
2024-10-27T00:53:57Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/versionStudio
2024-10-27T00:53:57Z [Deployment::InitializeConnectivity] Testing connectivity...
2024-10-27T00:53:57Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/versionStudio
2024-10-27T00:53:57Z [Deployment::InitializeConnectivity] Got https://setup.rbxcdn.com as the optimal base URL
2024-10-27T00:53:57Z [Bootstrapper::Run] Connectivity check finished
2024-10-27T00:53:57Z [Bootstrapper::GetLatestVersionInfo] Got channel as production
2024-10-27T00:53:57Z [Deployment::GetInfo] Getting deploy info for channel production
2024-10-27T00:53:57Z [HttpClientLoggingHandler::ProcessRequest] GET https://clientsettingscdn.roblox.com/v2/client-version/WindowsPlayer
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://clientsettingscdn.roblox.com/v2/client-version/WindowsPlayer
2024-10-27T00:53:58Z [RegistryKeyEx::SetValueSafe] Writing '' to HKEY_CURRENT_USER\SOFTWARE\ROBLOX Corporation\Environments\RobloxPlayer\Channel\www.roblox.com
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-rbxPkgManifest.txt
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-rbxPkgManifest.txt
2024-10-27T00:53:58Z [Bootstrapper::SetStatus] Applying Roblox modifications...
2024-10-27T00:53:58Z [Bootstrapper::ApplyModifications] Checking file mods...
2024-10-27T00:53:58Z [Bootstrapper::ApplyModifications] ClientSettings\ClientAppSettings.json already exists in the version folder, and is a match
2024-10-27T00:53:58Z [Bootstrapper::ApplyModifications] content\sounds\ouch.ogg was removed as a mod, restoring from content-sounds.zip
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] Downloading...
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] An exception occurred after downloading 0 bytes. (1/5)
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] An exception occurred after downloading 0 bytes. (2/5)
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] An exception occurred after downloading 0 bytes. (3/5)
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] An exception occurred after downloading 0 bytes. (4/5)
2024-10-27T00:53:58Z [Bootstrapper::DownloadPackage.content-sounds.zip] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
2024-10-27T00:53:58Z [HttpClientLoggingHandler::ProcessRequest] GET https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:59Z [HttpClientLoggingHandler::ProcessResponse] 200 OK https://setup.rbxcdn.com/version-4ffdeb3e393e469e-content-sounds.zip
2024-10-27T00:53:59Z [Bootstrapper::DownloadPackage.content-sounds.zip] An exception occurred after downloading 0 bytes. (5/5)
2024-10-27T00:53:59Z [Bootstrapper::DownloadPackage.content-sounds.zip] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
2024-10-27T00:53:59Z [LaunchHandler::LaunchRoblox] Bootstrapper task has finished
2024-10-27T00:53:59Z [LaunchHandler::LaunchRoblox] An exception occurred when running the bootstrapper
2024-10-27T00:53:59Z [App::FinalizeExceptionHandling] System.IO.DirectoryNotFoundException: Could not find a part of the path '%UserProfile%\AppData\Local\Bloxstrap\Downloads\cd77e0e77d698260809f8ae8b3993740'.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.DownloadPackage(Package package)
   at Bloxstrap.Bootstrapper.ApplyModifications()
   at Bloxstrap.Bootstrapper.Run()
   at Bloxstrap.Bootstrapper.Run()
```

**Comment by Wanwood1** [comment]
Please answer truthfully to all the above checkboxes.

**Comment by bluepilledgreat** [comment]
I'll look into this

**Comment by bluepilledgreat** [comment]
Does this happen every launch?

**Comment by pizzaboxer** [comment]
Fixed in 2.8.1 (assuming so, reopen if not)
